### PR TITLE
Add html notebook display for TreeSequence

### LINKED ISSUE #872: HTML summary for TreeSequence
When working in a jupyter notebook it would be nice to have a short summary of what's in the tree sequence. Basically, we want the numbers of things: trees, individuals, nodes, edges, etc. It'd be nice to get a HTML formatted table with each of these.

### PR DESCRIPTION
**Pull request by benjeffery**
## Description

Adds an html display in notebooks:
Screenshot from 2020-10-23 15-03-55

Fixes #872 

# PR Checklist:

- [x] Tests that fully cover new/changed functionality.
- [x] Documentation including tutorial content if appropriate.
- [x] Changelogs, if there are API changes.

### CODE REVIEW

**Comment by AdminBot-tskit** [comment]
📖 Docs for this PR can be previewed [here](https://tskit-dev.github.io/tskit-docs/923/)

**Comment by benjeffery** [comment]
@jeromekelleher I think it might be best to merge this now and fix the sizes once #871 is done.

File: python/tskit/trees.py [hunk 1]
```diff
@@ -3259,6 +3259,75 @@ def dump_text(
                 )
                 print(row, file=provenances)
 
+    def _repr_html_(self):
+        """
+        Called by jupyter notebooks to render a TreeSequence
+        """
+        table_rows = "".join(
```
Comment (jeromekelleher) [inline python/tskit/trees.py:3266]: If we're doing html stuff in the util module, maybe we could drop the implementation of this function in there too? It'll keep the line-count of trees.py down a bit and keep the html stuff in one place.

**Review by jeromekelleher — APPROVED** [review]
Merge away as you see fit - minor suggestion on code org.

File: python/tskit/trees.py (hunk 1 above)
Comment (benjeffery) [reply python/tskit/trees.py]: Yep, I've moved it there.

**Pull request merged**